src/utils/validator.ts

```diff
@@ -0,0 +1,95 @@
+import * as yup from 'yup';
+
+import { TFields } from './constants';
+import { granulatiry, periodData, TGranularity, TPeriod } from './period';
+
+const periods = Object.keys(periodData) as TPeriod[];
+periods.push('all');
+
+const sqlFuncs = ['SUM', 'AVG', 'COUNT'];
+
+const funcSchema = yup.string().test('func invalid', 'func invalid', value => {
+  return sqlFuncs.indexOf(value.toLocaleUpperCase()) > -1;
+});
+
+export const validateQueryWithGroupData = yup.object({
+  period: yup
+    .mixed<TPeriod>()
+    .required('Missing period parameter')
+    .oneOf(periods),
+  func: funcSchema,
+  col: yup.string().required('Missing col parameter'),
```

Comment:
Please validate `col`, it's still possible to provide custom SQL into col parameter

Reply:
![image](https://user-images.githubusercontent.com/69632363/127593397-a362570c-0999-4ca6-b401-9ca6f1b54815.png)
 updated